Task/35 decouple actions classes

prowwid opened the pull request:
| Question      | Answer
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Specs pass?   | yes
| Fixed issues  | [#35](https://github.com/php-kitchen/yii2-domain/issues/35)

prowwid committed:
#35: refactoring of base classes structure

----
Decouple Action base classes for a better flexibility
https://github.com/php-kitchen/yii2-domain/issues/35

prowwid committed:
#35: draft version of decoupled actions structure.

To achieve better hierarchy flexibility:
- split response, repository and session related actions to mixins
- extracted action hooks for successful and failed processing to mixin

New actions:
- RunCallable: for running strategies and callbacks
- RunService: for running services

----
Decouple Action base classes for a better flexibility
https://github.com/php-kitchen/yii2-domain/issues/35

prowwid committed:
#35: refactoring

Added new rendering methods to utilize controller's rendering functionality.

----
Decouple Action base classes for a better flexibility
https://github.com/php-kitchen/yii2-domain/issues/35

Djaconda commented on src/Web/Base/Actions/EntityModificationAction.php:
```diff
@@ -0,0 +1,133 @@
+    use ModelSearching;
+    use ResponseManagement;
+    use SessionMessagesManagement;
+    use EntityActionHooks;
```
The `EntityActionHooks` trait also containts `ResponseManagement`, `SessionMessagesManagement` it will cause methods collisions in php version less than 7.3. Could you please see example: 
http://sandbox.onlinephpfunctions.com/code/a44eb3506babe986e4c1b924967e7f0e22d77b32

Djaconda replied:
Also parent class has `ResponseManagement`, `SessionMessagesManagement`. It shouldn't cause any error but we have a warning in IDE about definition the same properties.

prowwid replied:
Thanks for identifying the issue:+1: Indeed I had 7.3 installed and didn't notice the issue
Fixed in the last commit.

Djaconda commented on src/Web/Base/Mixins/SessionMessagesManagement.php:
```diff
@@ -0,0 +1,82 @@
+        if (!$this->useFlashMessages || empty($message)) {
+            return;
+        }
+        $session = $this->serviceLocator->session;
```
Maybe you want to use `$this->getSession()` function?

prowwid replied:
Yeah:+1: 
It's an old piece of code that was missed in the past - fixed in the last commit:ok_hand:

prowwid committed:
#35: fixed methods collision error for PHP prior 7.3

----
Decouple Action base classes for a better flexibility
https://github.com/php-kitchen/yii2-domain/issues/35
